src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/WebhookGoCardlessServiceTest.java

```diff
@@ -0,0 +1,122 @@
+import static uk.gov.pay.directdebit.mandate.fixtures.GoCardlessPaymentFixture.aGoCardlessPaymentFixture;
+import static uk.gov.pay.directdebit.payments.fixtures.GoCardlessEventFixture.aGoCardlessEventFixture;
+import static uk.gov.pay.directdebit.payments.fixtures.TransactionFixture.aTransactionFixture;
+
+@RunWith(MockitoJUnitRunner.class)
+
+public class WebhookGoCardlessServiceTest {
+
+    @Mock
+    GoCardlessEventDao mockedGoCardlessEventDao;
+
+    @Mock
+    GoCardlessService mockedGoCardlessService;
+
+    @Mock
+    TransactionService mockedTransactionService;
+
+    WebhookGoCardlessService webhookGoCardlessService;
+    @Before
+    public void setUp() {
+        webhookGoCardlessService = new WebhookGoCardlessService(mockedGoCardlessEventDao, mockedGoCardlessService, mockedTransactionService);
+    }
+
+    @Test
+    public void shouldInsertEvents() {
+        GoCardlessEvent firstGoCardlessEvent = aGoCardlessEventFixture().toEntity();
+        GoCardlessEvent secondGoCardlessEvent = aGoCardlessEventFixture().toEntity();
+        List<GoCardlessEvent> events = Arrays.asList(firstGoCardlessEvent, secondGoCardlessEvent);
+        webhookGoCardlessService.handleEvents(events);
+        verify(mockedGoCardlessEventDao).insert(firstGoCardlessEvent);
+        verify(mockedGoCardlessEventDao).insert(secondGoCardlessEvent);
+    }
+
+    @Test
+    public void shouldHandlePaymentEventsWithAValidAction() {
+        GoCardlessEvent goCardlessEvent = aGoCardlessEventFixture().withResourceType("payments").withAction("paid_out").toEntity();
+        GoCardlessPayment goCardlessPayment = aGoCardlessPaymentFixture().toEntity();
+        Transaction transaction = aTransactionFixture().toEntity();
+        PaymentRequestEvent paymentRequestEvent = PaymentRequestEvent.paidOut(transaction.getPaymentRequestId());
+        when(mockedGoCardlessService.findPaymentForEvent(goCardlessEvent)).thenReturn(goCardlessPayment);
+        when(mockedTransactionService.findChargeFor(goCardlessPayment.getTransactionId())).thenReturn(transaction);
+        when(mockedTransactionService.paidOutFor(transaction)).thenReturn(paymentRequestEvent);
+
+        List<GoCardlessEvent> events = Collections.singletonList(goCardlessEvent);
+        webhookGoCardlessService.handleEvents(events);
+        verify(mockedGoCardlessEventDao).insert(goCardlessEvent);
+        verify(mockedGoCardlessEventDao).updatePaymentRequestEventId(goCardlessEvent.getId(), paymentRequestEvent.getId());
+    }
+
+    @Test
+    public void shouldNotHandleMandateEventsWithAnInvalidResource() {
+        GoCardlessEvent goCardlessEvent = aGoCardlessEventFixture().withResourceType("not_handled").withAction("created").toEntity();
+
+        List<GoCardlessEvent> events = Collections.singletonList(goCardlessEvent);
+        webhookGoCardlessService.handleEvents(events);
+
+        verify(mockedGoCardlessEventDao).insert(goCardlessEvent);
```

Comment:
A `verifyNoMoreInteractions(mockedGoCardlessEventDao)` may be beneficial here (applies to the other `shouldNotHandle` tests too)